src/search/FindUtils.js

```diff
@@ -352,11 +387,209 @@ define(function (require, exports, module) {
+    }
+
+    /**
+     * Notifies that a node has finished indexing the files
+     */
+    function notifyIndexingFinished() {
+        indexingInProgress = false;
+        exports.trigger(exports.SEARCH_INDEXING_FINISHED);
+    }
+
+    /**
+     * Return true if indexing is in pregress in node
+     * @returns {boolean} true if files are being indexed in node
+     */
+    function isIndexingInProgress() {
+        return indexingInProgress;
+    }
+
+    /**
+     * Set if we need to collapse all results in the results pane
+     * @param {boolean} collapse true to collapse
+     */
+    function setCollapseResults(collapse) {
+        collapseResults = collapse;
+        exports.trigger(exports.SEARCH_COLLAPSE_RESULTS);
+    }
+
+    /**
+     * check if results should be collapsed
+     * @returns {boolean} true if results should be collapsed
+     */
+    function isCollapsedResults() {
+        return collapseResults;
+    }
+
     exports.parseDollars                    = parseDollars;
     exports.getInitialQuery                 = getInitialQuery;
     exports.hasCheckedMatches               = hasCheckedMatches;
     exports.performReplacements             = performReplacements;
     exports.labelForScope                   = labelForScope;
     exports.parseQueryInfo                  = parseQueryInfo;
+    exports.prioritizeOpenFile              = prioritizeOpenFile;
+    exports.getOpenFilePath                 = getOpenFilePath;
+    exports.setNodeSearchDisabled           = setNodeSearchDisabled;
+    exports.isNodeSearchDisabled            = isNodeSearchDisabled;
+    exports.setInstantSearchDisabled        = setInstantSearchDisabled;
+    exports.isInstantSearchDisabled         = isInstantSearchDisabled;
+    exports.isNodeSearchInProgress          = isNodeSearchInProgress;
+    exports.isIndexingInProgress            = isIndexingInProgress;
+    exports.setCollapseResults              = setCollapseResults;
+    exports.isCollapsedResults              = isCollapsedResults;
     exports.ERROR_FILE_CHANGED              = "fileChanged";
+
+    // event notification functions
+    exports.notifyFileFiltersChanged        = notifyFileFiltersChanged;
+    exports.notifySearchScopeChanged        = notifySearchScopeChanged;
+    exports.notifyNodeSearchStarted         = notifyNodeSearchStarted;
+    exports.notifyNodeSearchFinished        = notifyNodeSearchFinished;
+    exports.notifyIndexingStarted           = notifyIndexingStarted;
+    exports.notifyIndexingFinished          = notifyIndexingFinished;
+
+    // events raised by FindUtils
+    exports.SEARCH_FILE_FILTERS_CHANGED              = "fileFiltersChanged";
+    exports.SEARCH_SCOPE_CHANGED                     = "searchScopeChanged";
+    exports.SEARCH_INDEXING_STARTED                  = "searchIndexingStarted";
+    exports.SEARCH_INDEXING_FINISHED                 = "searchIndexingFinished";
+    exports.SEARCH_COLLAPSE_RESULTS                  = "searchCollapseResults";
 });
```

Comment:
Are we listening to change in the instantSearch enable/disable preference by user?